%% The text of your abstract and nothing else (other than comments) goes here.
%% The rest of the text that is supposed to go on the abstract page will be 
%% generated by the abstractpage environment.
%% This file should be \input (not \include 'd) from cover.tex.

\noindent{}New sources of data, such as next generation sequencing (NGS) of pathogen ge\-nomes, electronic medical records (EMR), and omics assays like RNA-seq and mass cytometry, are poised to transform clinical infectious diseases. One of the greatest challenges in applying these ``big data'' toward clinical practice is the development of bioinformatics techniques and software that make downstream analyses routine, rigorous, and actionable. Herein, we develop software and integrative statistical models for several combinations of these data to address urgent global threats in infectious diseases, including the spread of healthcare associated infections (HAIs), skyrocketing rates of antimicrobial resistance, and understanding human immune responses to poorly characterized mosquito-borne viruses.

We first demonstrate that \emph{de novo} assembly of long reads can finish the genomes of hospital bacterial isolates with sufficient resolution for discovering a sin\-gle-\allowbreak nu\-cle\-o\-tide variant that enabled resistance emerging during failed antimicrobial therapyâ€”in our case, the development of quinolone resistance in a newly sequenced strain of \emph{Stenotrophomonas maltophilia}. We then describe a new genome browser and modular software suite that facilitate the routine usage of NGS by the Pathogen Surveillance Program at The Mount Sinai Hospital for reconstructing HAI transmission networks between patients. These software additionally provide actionable visualizations of genomic data tailored to infection control physicians. The use of long read data for hospital surveillance is novel and offers unique opportunities to capture recombination and horizontal transfer events occuring throughout the rapid evolution of virulence and resistance in HAI strains. 

To convert these data into management strategies, we use machine learning algorithms on seven years of EMR data at Mount Sinai to precisely estimate per-patient costs for \emph{Clostridium difficile} infection, helping to anchor a cost-benefit analysis for selecting interventions and surveillance investments. Finally, in a more futuristic study, we integrate RNA-seq, mass cytometry, and multiplexed immunoassay data to comprehensively profile the human immune response to chikungunya virus, a re-emerging arthritogenic arbovirus, which reveals a significant role for monocytes and novel biomarkers for clinical outcomes like symptom severity and immunogenicity. Because of the many applications for multiscale analysis we establish in this dissertation, we conclude that it is indeed transforming infectious diseases.